griptape/memory/task/storage/text_artifact_storage.py

```diff
@@ -2,15 +2,18 @@
 from typing import TYPE_CHECKING, Any, Optional
 from attrs import define, field
 from griptape.artifacts import TextArtifact, BaseArtifact, ListArtifact
+from griptape.drivers import BaseVectorStoreDriver
+from griptape.engines.rag import RagEngine, RagContext
 from griptape.memory.task.storage import BaseArtifactStorage
 
 if TYPE_CHECKING:
-    from griptape.engines import BaseSummaryEngine, CsvExtractionEngine, JsonExtractionEngine, VectorQueryEngine
+    from griptape.engines import BaseSummaryEngine, CsvExtractionEngine, JsonExtractionEngine
 
 
 @define
 class TextArtifactStorage(BaseArtifactStorage):
-    query_engine: VectorQueryEngine = field(kw_only=True)
+    vector_store_driver: BaseVectorStoreDriver = field(kw_only=True)
+    rag_engine: RagEngine = field(kw_only=True)
```

Comment:
Should `rag_engine` should be `Optional` similar to the other Engines?

Maybe it could do [something similar](https://github.com/griptape-ai/griptape/pull/860/files#diff-0da0341f92dc228cc857f38275578abf45451919ce1389f021930b9b463b0e6eR45) to `VectorStoreClient` if the engine is not set?